lldb/test/API/lang/swift/async/stepping/step_out/TestSteppingOutAsync.py

```diff
@@ -0,0 +1,44 @@
+import lldb
+from lldbsuite.test.decorators import *
+import lldbsuite.test.lldbtest as lldbtest
+import lldbsuite.test.lldbutil as lldbutil
+import re
+
+
+class TestCase(lldbtest.TestBase):
+
+    def check_and_get_frame_names(self, process):
+        frames = process.GetSelectedThread().frames
+        # Expected frames:
+        # ASYNC___0___ <- ASYNC___1___ <- ASYNC___2___ <- ASYNC___3___ <- Main
+        num_frames = 5
+        func_names = list(map(lambda frame: frame.GetFunctionName(), frames[:num_frames]))
+        for idx in range(num_frames - 1):
+            self.assertRegex(func_names[idx], f"ASYNC___{idx}___")
+        self.assertRegex(func_names[num_frames - 1], "Main")
+        return func_names
+
+    def step_out_checks(self, thread, expected_func_names):
+        # Keep stepping out, comparing the top frame's name with the expected name.
+        for expected_func_name in expected_func_names:
+            error = lldb.SBError()
+            thread.StepOut(error)
+            self.assertSuccess(error, "step out failed")
+            stop_reason = thread.GetStopReason()
+            self.assertStopReason(stop_reason, lldb.eStopReasonPlanComplete)
+            self.assertEqual(thread.frames[0].GetFunctionName(), expected_func_name)
+
+    @swiftTest
+    @skipIf(oslist=["windows", "linux"])
+    @skipIf("rdar://133849022")
```

Comment:
Given the PR description:
> This PR adds tests demonstrating where the debugger fails

should this be an expectedFailure?

Reply:
I was debating this but we would risk having sporadic xpasses: if the stars align and the "heap cfas" numeric values are such that certain StackIDs comparisons behave in some pattern, the test _could_ pass.